colossalai/inference/tensor_parallel/modeling/llama.py

```diff
@@ -139,6 +126,21 @@ def llama_model_forward(
                 # infer_state.decode_value_buffer = torch.empty((batch_size, self.tp_head_num_, self.head_dim_), dtype=torch.float16, device="cuda")
                 infer_state.block_loc[:, seq_length_with_past - 1] = infer_state.decode_mem_index
 
+        if infer_state.is_context_stage:
```

Comment:
Swap codes between 129~142 lines  and codes between144~152 lines to avoid redundant code, “position_ids = None”